test/integration.test.js

```diff
@@ -55,4 +68,39 @@ describe('Integration testing generateFromFile() to make sure the result of the
     const file = await readFile(path.join(outputDir, testOutputFile), 'utf8');
     expect(file).toMatchSnapshot();
   });
+
+  it('should ignore specified files with noOverwriteGlobs', async () => {
+    const logSpyDebug = jest.spyOn(log, 'debug').mockImplementation(() => {});
```

Comment:
why not `log.debug = jest.fn();` and then `expect(log.debug)`, then you do not need to do `mockRestore()` imho

Reply:
I tried that. jest.fn() does not work but spyOn() does. The former does not actually track the log.debug() calls in other files, but the latter does by creating a [layer of observation around the existing methods](https://medium.com/javascript-journal-unlocking-project-potential/jest-fn-vs-jest-spyon-understanding-the-differences-for-effective-testing-0f5928f7a411).